### PR TITLE
Refactor/competition

### PR DESCRIPTION
**Pull request by Adentalis**

### CODE REVIEW

File: src/matchmaker/drawing.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  pairPlayersRoundOne
+} = require("./drawingAlgorithms.js");
+
+const { createMatches } = require("./match.js");
+
+// drawRound : [players] -> [matches]
+function drawRound({ players }) {
+  const isFirstRound = players[0].matchIds.length == 0;
+
+  if (!isFirstRound) {
```
Comment (mrvary) [inline src/matchmaker/drawing.js]: it seems more clear when you don't negate it. Also you could simplify it by using a ternary expression for getting the matches and then just returning them, like so: `const pairings = players[0].matchIds.length === 0 ? drawFirstRound(players) : drawLaterRound(players); return createMatches(pairings);` (prettier should then move this to multiple lines)

File: src/matchmaker/drawingAlgorithms.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        const second = Math.floor(Math.random() * array.length);
+        [array[first], array[second]] = [array[second], array[first]];
+    }
+    return array;
+}
+
+
+
+module.exports = {
+    // pubic
```
Comment (mrvary) [inline src/matchmaker/drawingAlgorithms.js]: `public` ;)

File: src/matchmaker/drawingAlgorithms.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+function shuffle(array) {
+    const TIMES = array.length;
+    for (let i = 0; i < TIMES; i++) {
+        const first = Math.floor(Math.random() * array.length);
+        const second = Math.floor(Math.random() * array.length);
+        [array[first], array[second]] = [array[second], array[first]];
+    }
+    return array;
+}
+
```
Comment (mrvary) [inline src/matchmaker/drawingAlgorithms.js]: we need to have another look at your style setup, files don't need to contain multiple empty lines

File: src/matchmaker/drawingAlgorithms.js (hunk 2 above)
Comment (mrvary) [reply src/matchmaker/drawingAlgorithms.js]: also what's your reasoning for putting all those player functions into `drawingAlgorithms` ?

Comment (Adentalis) [reply src/matchmaker/drawingAlgorithms.js]: These functions are just used once for the drawing of the first round. So i removed them from the player itself and put them to this new file (maybe you have an idea for a better name then drawingAlgorithms), where all the necessary drawing functions are together in one place.

File: src/matchmaker/drawing.js (hunk 1 above)
Comment (Adentalis) [reply src/matchmaker/drawing.js]: good point

File: src/matchmaker/player.js [hunk 4]
```diff
@@ -22,97 +34,26 @@ function createPlayer(dataFromJSON) {
...
-    players.push(player2);
-  }
-
-  return players;
-}
+// ToDo
+function updatePlayers() {}
 
 module.exports = {
   // pubic
```
Comment (mrvary) [inline src/matchmaker/player.js]: :)

File: src/matchmaker/drawingAlgorithms.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+  for (let i = 0; i < TIMES; i++) {
+    const first = Math.floor(Math.random() * array.length);
+    const second = Math.floor(Math.random() * array.length);
+    [array[first], array[second]] = [array[second], array[first]];
+  }
+  return array;
+}
+
+module.exports = {
+  // pubic
```
Comment (mrvary) [inline src/matchmaker/drawingAlgorithms.js]: :)

File: src/matchmaker/drawingAlgorithms.js (hunk 2 above)
Comment (mrvary) [reply src/matchmaker/drawingAlgorithms.js]: 👍

**Pull request merged**